src/dl_node.cpp

```diff
@@ -30,132 +31,36 @@ namespace ovms {
-        notifyEndQueue.push(*this);
-        return status;
-    }
-    return status;
-}
-
-Status DLNode::requestExecuteRequiredResources() {
-    Status status = StatusCode::OK;
-    status = this->modelManager.getModelInstance(
-        this->modelName,
-        this->modelVersion.value_or(0),
-        this->model,
-        this->modelUnloadGuard);
-
-    if (!status.ok()) {
-        SPDLOG_DEBUG("Getting modelInstance failed for node: {} with: {}", getName(), status.string());
-        return status;
-    }
-
-    status = prepareInputsAndModelForInference();
-    if (!status.ok()) {
-        return status;
-    }
-    auto& inferRequestsQueue = this->model->getInferRequestsQueue();
-    this->nodeStreamIdGuard = std::make_unique<NodeStreamIdGuard>(inferRequestsQueue);
-    return status;
+Status DLNode::execute(session_key_t sessionKey, PipelineEventQueue& notifyEndQueue) {
+    auto& nodeSession = getNodeSession(sessionKey);                   // TODO error check?
+    auto& dlNodeSession = dynamic_cast<DLNodeSession&>(nodeSession);  // TODO error check dynamic_cast
+    return dlNodeSession.execute(notifyEndQueue, WAIT_FOR_STREAM_ID_TIMEOUT_MICROSECONDS, *this);
 }
 
-Status DLNode::setInputsForInference(InferenceEngine::InferRequest& infer_request) {
-    Status status = StatusCode::OK;
-    try {
-        // Prepare inference request, fill with input blobs
-        for (const auto& kv : this->inputBlobs) {
-            std::string realModelInputName;
-            if (!getRealInputName(kv.first, &realModelInputName).ok()) {
-                SPDLOG_WARN("DLNode::{} [Node name: {}]; cannot find real model input name for alias: {}", __FUNCTION__, getName(), kv.first);
-                return StatusCode::INTERNAL_ERROR;
-            }
-            infer_request.SetBlob(realModelInputName, kv.second);
-        }
-        // OV implementation the InferenceEngineException is not
-        // a base class for all other exceptions thrown from OV.
-        // OV can throw exceptions derived from std::logic_error.
-    } catch (const InferenceEngine::details::InferenceEngineException& e) {
-        status = StatusCode::OV_INTERNAL_DESERIALIZATION_ERROR;
-        SPDLOG_DEBUG("[Node: {}] {}; exception message: {}", getName(), status.string(), e.what());
-    } catch (std::logic_error& e) {
-        status = StatusCode::OV_INTERNAL_DESERIALIZATION_ERROR;
-        SPDLOG_DEBUG("[Node: {}] {}; exception message: {}", getName(), status.string(), e.what());
-    } catch (...) {
-        status = StatusCode::OV_INTERNAL_DESERIALIZATION_ERROR;
-        SPDLOG_DEBUG("[Node: {}] {}; with unknown exception", getName(), status.string());
-    }
+Status DLNode::fetchResults(NodeSession& nodeSession, SessionResults& nodeSessionOutputs) {
+    auto& dlNodeSession = dynamic_cast<DLNodeSession&>(nodeSession);
+    const auto& sessionMetadata = nodeSession.getNodeSessionMetadata();
+    std::pair<NodeSessionMetadata, BlobMap> sessionResults{sessionMetadata, {}};
```

Comment:
Maybe it might be good idea to have typedef for `std::pair<NodeSessionMetadata, BlobMap>`? For example `SessionResult`.

Reply:
This would be confusing since one node session can spawne several std::pair<NodeSessionMetadata, BlobMap>. It is not visible at this point since we do not have demultiplexer implemented yet.

Reply:
After a bit of thought I agree. Added with next commit.